compiler/rustc_parse/src/parser/item.rs

```diff
@@ -2385,7 +2385,20 @@ impl<'a> Parser<'a> {
             };
             let (pat, ty) = if is_name_required || this.is_named_param() {
                 debug!("parse_param_general parse_pat (is_name_required:{})", is_name_required);
-
+                if let token::Ident(symname, _) = this.token.kind
+                    && this.prev_token.is_lifetime()
```

Comment:
It's not *critical*, but we should be checking `self.may_recover()`, just in case.

Reply:
Sorry. I didn't find this function `self.may_recover()`. Does that mean we should check if we can recover from the parsing error? I don't think so, because `'a Sized` is not a legal syntax. The parser only check if the code missing `'` between some words